### ISSUE TITLE
Should Integral(f(x), (x, 0, 1)) have is_number = False? 

### ISSUE THREAD

**Issue by normalhuman** [sympy/sympy#13847]
According to #13619, `is_number` now is only True for something that will evaluate to a number (though it may not do so with precision). For example, `f(2).is_number == False` if `f` is an undefined function. However, 
```
from sympy import *
x, f = Symbol('x'), Function('f')
Integral(f(x), (x, 0, 1)).is_number  
```
returns True in current master branch. It seems to me that the definite integral of an undefined function `f` is no more numeric than `f(1)`. 

Should `is_number` mean: no free symbols __and__  no bound symbols __and__ no AppliedUndef?

**Comment by abaokar-07** [comment]
@normalhuman you are right, because if f(x) is not defined, then it's integral will lead to another function say g(x) and then after putting limits we will have [g(1) - g(0)]. So ultimately we should get False

**Comment by samyak-jain** [comment]
Can I take up this issue?

**Comment by asmeurer** [comment]
Agreed. is_number already seems to mean "can be evalfed", so in line with that PR, `Integral(f(x), (x, 0, 1)).is_number` should match `f(0).is_number`.
